censys/asm/client.py

```diff
@@ -25,3 +25,18 @@ def __init__(self, api_key: Optional[str] = None, **kwargs):
         self.events = Events(api_key, **kwargs)
         self.clouds = Clouds(api_key, **kwargs)
         self.risks = Risks(api_key, **kwargs)
+
+        # Save the arguments for parameterized client usage
+        self.__api_key = api_key
```

Comment:
I think instead of saving the api keys twice we could just access `_api_key` from
`self.domains._api_key`.